fast_llm/data/dataset/gpt/memmap.py

```diff
@@ -1,8 +1,10 @@
+import io
```

Comment:
Reworked this file so the various concepts are properly compartmentalized. Next step would be to extract those components into separate classes, it's not strictly needed yet but will become if we keep adding stuff.